scripts/pyenv_setup.sh

```diff
@@ -0,0 +1,119 @@
+#!/bin/bash
+# This script correctly sets up pyenv
+#
+# Assumptions:
+# - This script assumes you're calling from the top directory of the repository
+set -eu
+
+# Check if a command is available
+require() {
+    command -v "$1" >/dev/null 2>&1
+}
+
+query_big_sur() {
+    if require sw_vers && sw_vers -productVersion | grep -E "11\." > /dev/null; then
+        return 0
+    fi
+    return 1
+}
+
+get_shell_startup_script() {
+  local _startup_script=''
+  if [ -n "$SHELL" ]; then
+    case "$SHELL" in
+      */bash)
+        _startup_script="${HOME}/.bash_profile"
+        ;;
+      */zsh)
+        _startup_script="${HOME}/.zshrc"
+        ;;
+      */fish)
+        _startup_script="${HOME}/.config/fish/config.fish"
+        ;;
+      *)
+        echo "$SHELL is currently not supported."
+        exit 1
+    esac
+  else
+    echo "The environment variable \$SHELL needs to be defined."
+    exit 1
+  fi
+  echo "$_startup_script"
+}
+
+_append_to_startup_script() {
+  if [ -n "$SHELL" ]; then
+    case "$SHELL" in
+      */bash)
+        # shellcheck disable=SC2016
+        echo -e '\nif command -v pyenv 1>/dev/null 2>&1; then\n  eval "$(pyenv init -)"\nfi' >> "${1}"
+        ;;
+      */zsh)
+        # shellcheck disable=SC2016
+        echo -e '\nif command -v pyenv 1>/dev/null 2>&1; then\n  eval "$(pyenv init -)"\nfi' >> "${1}"
+        ;;
+      */fish)
+        echo -e '\n\n# pyenv init\nif command -v pyenv 1>/dev/null 2>&1\n  pyenv init - | source\nend' >> "$1"
+    esac
+
+    echo "  --> Tail of ${1}"
+    tail -n 3 "${1}"
+  fi
+}
+
+append_to_config() {
+  if [ -n "$1" ]; then
+    echo "Adding pyenv init (if missing) to ${1}..."
+    # shellcheck disable=SC2016
+    if ! grep -qF 'eval "$(pyenv init -)"' "${1}"; then
```

Comment:
Probably just grep for `pyenv init` will suffice, since this isn't correct for what you're appending for `fish`.